src/main/java/core/basesyntax/strategy/PurchaseOperation.java

```diff
@@ -0,0 +1,19 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.Inventory;
+
+public class PurchaseOperation implements FruitOperationHandler {
+    @Override
+    public void executeOperation(FruitTransaction transaction, Inventory inventory) {
+        int currentBalance = inventory.getInventory().getOrDefault(transaction.getFruit(), 0);
+        int newBalance = currentBalance - transaction.getQuantity();
+
+        if (newBalance < 0) {
+            throw new RuntimeException(
+                    "Cannot complete purchase: insufficient stock for " + transaction.getFruit());
+        }
+
+        inventory.getInventory().put(transaction.getFruit(), newBalance);
```

Comment:
Directly modifying the inventory map here violates encapsulation. It would be better to have a method in `Inventory` that updates the stock, which could be called here instead of modifying the map directly.